snuba/manual_jobs/runner.py

```diff
@@ -34,7 +63,69 @@ def _build_job_spec_from_entry(content: Any) -> JobSpec:
     return job_spec
 
 
+def _job_lock_key(job_id: str) -> str:
+    return f"snuba:manual_jobs:{job_id}:lock"
+
+
+def _job_status_key(job_id: str) -> str:
+    return f"snuba:manual_jobs:{job_id}:execution_status"
+
+
+def _acquire_job_lock(job_id: str) -> bool:
+    return bool(
+        _redis_client.set(
+            name=_job_lock_key(job_id), value=1, nx=True, ex=(24 * 60 * 60)
+        )
+    )
+
+
+def _release_job_lock(job_id: str) -> None:
+    _redis_client.delete(_job_lock_key(job_id))
```

Comment:
I think that's overkill for the level of synchronization we're looking for here. We're basically just looking to prevent mistakes of e.g. a single engineer clicking twice because they think the first click did nothing. We don't have to worry about 1000 users trying to run the same job concurrently.